### ISSUE TITLE
Update template to latest Bootstrap (with dark mode)

### ISSUE THREAD

**Issue by jfy133** [MultiQC/MultiQC#1698]
### Description of feature

It would make it easier on my eyes ;)

As far as I can tell most colours should still be 'OK'(tm), it would mainly be flipping the white background to black, and black text to white. Only additional thing would be that the you would need a dark-mode version of the MultiQC logo - something I'm not sure how easy it would be.

In addition, this could be supported by a toggle switch, and ideally a configuration file boolean.

**Comment by jfy133** [comment]
(Font is slightly better with grey colour #999)

**Comment by ewels** [comment]
Yup, I would love this too. MultiQC is built with Bootstrap and that has native dark mode support coming soon-ish: https://twitter.com/getbootstrap/status/1494101332722024449

I'm hoping to do some major renovation of the front end template, so this can be part of that project. Upgrade bootstrap and support dark mode in one 🚀

**Comment by ewels** [comment]
Until then, I think Dark Reader does a pretty good job..? https://darkreader.org/

**Comment by ewels** [comment]
Next Bootstrap release should include dark mode support, so we're close now.

Several parts to this issue:

- Updating the template to use the new bootstrap version (and updating class names etc)
- Adding functionality to detect / switch between dark mode and light mode
- Ensuring that all report elements work well in both modes, or can switch themes
    - Plots is the big one of course. It may make sense to wait for this before going ahead: https://github.com/ewels/MultiQC/issues/1789

**Comment by vladsavelyev** [comment]
It sounds like it would be a rewrite, rather than an upgrade:
https://www.reddit.com/r/webdev/comments/1ajjr5m/migrating_bootstrap_3x_to_5x/
https://getbootstrap.com/docs/4.0/migration/
https://getbootstrap.com/docs/5.0/migration/

Not sure how dark mode is worth it, but I'd put it in the priority behind the intermediate format and many other issues.

**Comment by ewels** [comment]
Bookmarking this stack-overflow [post](https://stackoverflow.com/a/71726400) for myself. Sounds like we will need to export the themes from Python into a JS object so that we have the variables available in the front end to be able to switch dynamically. 

```python
fig_plotly_dark = go.FigureWidget(fig)
fig_plotly_dark.layout.template = 'plotly_dark'
fig_plotly_dark.to_plotly_json()["layout"]["template"]
```
